Server/mods/deathmatch/logic/luadefs/CLuaTeamDefs.cpp

```diff
@@ -67,282 +65,86 @@ void CLuaTeamDefs::AddClass(lua_State* luaVM)
-
-    lua_pushboolean(luaVM, false);
-    return 1;
+    return team->GetFriendlyFire();
 }
 
-int CLuaTeamDefs::GetTeamFriendlyFire(lua_State* luaVM)
+std::vector<CPlayer*> CLuaTeamDefs::GetPlayersInTeam(CTeam* team)
 {
-    CTeam* pTeam;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pTeam);
-
-    if (!argStream.HasErrors())
-    {
-        bool bFriendlyFire;
-        if (CStaticFunctionDefinitions::GetTeamFriendlyFire(pTeam, bFriendlyFire))
-        {
-            lua_pushboolean(luaVM, bFriendlyFire);
-            return 1;
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
-
-    lua_pushboolean(luaVM, false);
-    return 1;
+    return team->GetPlayers();
 }
 
-int CLuaTeamDefs::CountPlayersInTeam(lua_State* luaVM)
+std::uint32_t CLuaTeamDefs::CountPlayersInTeam(CTeam* team) noexcept
 {
-    CTeam* pTeam;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pTeam);
-
-    if (!argStream.HasErrors())
-    {
-        unsigned int uiCount;
-        if (CStaticFunctionDefinitions::CountPlayersInTeam(pTeam, uiCount))
-        {
-            lua_pushnumber(luaVM, uiCount);
-            return 1;
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
-
-    lua_pushboolean(luaVM, false);
-    return 1;
+    return team->CountPlayers();
 }
 
-int CLuaTeamDefs::SetTeamName(lua_State* luaVM)
+bool CLuaTeamDefs::SetPlayerTeam(CPlayer* player, CTeam* team) noexcept
 {
-    CTeam*  pElement;
-    SString strName;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pElement);
-    argStream.ReadString(strName);
-
-    if (!argStream.HasErrors())
-    {
-        if (CStaticFunctionDefinitions::SetTeamName(pElement, strName))
-        {
-            lua_pushboolean(luaVM, true);
-            return 1;
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
-
-    lua_pushboolean(luaVM, false);
-    return 1;
+    return CStaticFunctionDefinitions::SetPlayerTeam(player, team);
 }
 
-int CLuaTeamDefs::SetTeamColor(lua_State* luaVM)
+bool CLuaTeamDefs::SetTeamName(CTeam* team, const std::string_view name) noexcept
 {
-    CTeam*        pElement;
-    unsigned char ucRed;
-    unsigned char ucGreen;
-    unsigned char ucBlue;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pElement);
-    argStream.ReadNumber(ucRed);
-    argStream.ReadNumber(ucGreen);
-    argStream.ReadNumber(ucBlue);
-
-    if (!argStream.HasErrors())
-    {
-        if (CStaticFunctionDefinitions::SetTeamColor(pElement, ucRed, ucGreen, ucBlue))
-        {
-            lua_pushboolean(luaVM, true);
-            return 1;
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
-
-    lua_pushboolean(luaVM, false);
-    return 1;
+    return CStaticFunctionDefinitions::SetTeamName(team, name.data());
```

Comment:
1. Is `CStaticFunctionDefinitions::SetTeamName` exception safe? - most likely not, remove `noexcept`
2. Are you sure `name` is always null-terminated?

Reply:
GitHub kinda messed up the comment on this, just ignore the deleted lines.